praw/models/reddit/subreddit.py

```diff
@@ -2155,3 +2174,103 @@ def revisions(self, **generator_kwargs):
         url = API_PATH['wiki_revisions'].format(subreddit=self.subreddit)
         return WikiPage._revision_generator(
             self.subreddit, url, generator_kwargs)
+
+
+class SubredditEmoji(object):
+    """Provides a set of functions to a Subreddit for emoji."""
+
+    def __getitem__(self, name):
+        """Lazily return the Emoji for the subreddit named ``name``.
+
+        This method is to be used to fetch a specific emoji url, like so:
+
+        .. code:: python
+
+           emoji = reddit.subreddit('iama').emoji['cake']
+           print(emoji)
+
+        """
+        if e.name == name for e in self.subreddit.emoji:
+            return e for e in self.subreddit.emoji if e.name == name
+        else:
+            return None
+
+    def __init__(self, subreddit):
+        """Create a SubredditEmoji instance.
+
+        :param subreddit: The subreddit whose emoji are affected.
+        :param reddit: The reddit instance.
+
+        """
+        self.subreddit = subreddit
+        self.reddit = self.subreddit._reddit
+
+    def __iter__(self):
+        """Iterate through the Emoji for the subreddit.
+
+        This method is to be used to discover all emoji for a subreddit:
+
+        .. code:: python
+
+           for emoji in reddit.subreddit('praw_test').emoji:
+               print(emoji)
+
+        """
+        response = self.subreddit._reddit.get(
+            API_PATH['emoji_list'].format(subreddit=self.subreddit))
+        for emoji_name, emoji_data in \
+                response[self.subreddit.fullname].items():
+            yield Emoji(self.subreddit._reddit,
+                        self.subreddit, emoji_name, _data=emoji_data)
+
+    def add(self, name, filepath):
+        """Add an emoji to this subreddit.
+
+        :param filepath: Path to the file being added.
+        :returns: The Emoji added.
+
+        To add ``'cake'`` to the subreddit ``'praw_test'`` try:
+
+        .. code:: python
+
+           reddit.subreddit('praw_test').emoji.add('cake','cake.png')
+
+        """
+        filepath = filepath.strip()
+        filebasename = os.path.basename(filepath)
+        data = {'filepath': filebasename, 'mimetype': 'image/jpeg'}
+        if filebasename.lower().endswith('.png'):
+            data['mimetype'] = 'image/png'
+        url = API_PATH['emoji_lease'].format(
+            subreddit=self.subreddit, method='add')
```

Comment:
I thought bboe commented on this, but I can't seem to find it. What does `method='add'` do?